### PR TITLE
Mobile app interface changes

### LINKED ISSUE #9005: Mobile api keys design review
@javitonino commented on [Fri Jul 15 2016](https://github.com/CartoDB/cartodb-central/issues/1320)

 - [ ] Api key nested too deep
 - [x] Remove user key (mobile SDK is going to stop using it)
 - [x] Allow update dev -> open/closed (done in Central/backend, missing UI and Box controller changes)

Hey @josecruz:
After meeting with @jaakla to prepare the launch of the CartoSDK, he proposed some changes to the design. Let's meet when you are available and I'll give you the details, please :pray: 

---

@jaakla commented on [Fri Jul 15 2016](https://github.com/CartoDB/cartodb-central/issues/1320#issuecomment-232953405)

Also please remove Carto apiKey from this screen, as we cannot use it in mobile at all due to security reasons.

---

@josecruz commented on [Wed Jul 20 2016](https://github.com/CartoDB/cartodb-central/issues/1320#issuecomment-233909678)

Hey! I upload here last changes:
api keys - 00 - 02 - edit form

### PR DESCRIPTION
**Pull request by MariaCheca**
Closes #9005 

 - [x] Remove user key (mobile SDK is going to stop using it)
 - [x] Allow update dev -> open/closed (done in Central/backend, missing UI and Box controller changes)

edit mobile application carto

**UPDATE** Modal before change app type:
screen shot 2016-07-28 at 12 51 31

CR @javitonino @javierarce

### CODE REVIEW

**Comment by javitonino** [comment]
We also need to modify the `update` controller to allow updating the app_type. Do you prefer to do it yourself, or should I?

**Comment by MariaCheca** [comment]
@javitonino I'll do it, don't worry

File: app/controllers/carto/admin/mobile_apps_controller.rb [hunk 1]
```diff
@@ -71,10 +71,11 @@ def create
   end
 
   def update
-    updated_attributes = params[:mobile_app].symbolize_keys.slice(:name, :description, :icon_url)
+    updated_attributes = params[:mobile_app].symbolize_keys.slice(:name, :description, :icon_url, :app_type)
     @mobile_app.name = updated_attributes[:name]
     @mobile_app.icon_url = updated_attributes[:icon_url]
     @mobile_app.description = updated_attributes[:description]
+    @mobile_app.app_type = updated_attributes[:app_type] if @mobile_app.app_type === 'dev'
```
Comment (javitonino) [inline app/controllers/carto/admin/mobile_apps_controller.rb]: This works, but actually does no validation in the box. If you see below, the actual info send to Central is updated attributes. This means that it does no actual validation of the app_type change in the box side (Central does it, 
though). It should be fine as the UI locks you from changing it, and it is validated in Central anyway, but you may want to add actual validation.

This assignment to @mobile_app is used to run the validation and to maintain the updated values in the UI if the validation fails, so I think the assignment should always be done.

PS: In ruby, we use `==`, not triple equal.

**Comment by javierarce** [comment]
👍  to the HTML part

**Comment by javitonino** [comment]
I made the backend change myself (trying to deploy today).

Buuuuut, I forgot to mention we are also supposed to include a modal when upgrading from dev to open, to say that the License key for the app is going to be regenerated (recycling the regenerate api keys modal). Sorry :(

Also, "Your API keys" (just on top of the license key textarea) should read "Application key" according to the design.

**Comment by MariaCheca** [comment]
Oooh ok, I should have had that in mind. Don't worry, when I have it ready I'll push it with a screenshot :)

**Comment by javitonino** [comment]
@MariaCheca Can you add a short notice to the modal saying something like, "The license key will change and you will need to update it in your app." I'd ask Christa, but she's out for holidays.

**Comment by MariaCheca** [comment]
✨

**Pull request merged**